### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by laisecf**

### CODE REVIEW

File: pyinterprod/pronto/goa.py [hunk 1]
```diff
@@ -74,8 +73,22 @@ def import_annotations(ora_url: str, pg_url: str):
             """
         )
 
-        sql = "INSERT INTO protein2go VALUES %s"
-        execute_values(pg_cur, sql, ora_cur, page_size=1000)
+        records = []
+        sql = """
+              INSERT INTO protein2go (protein_acc, term_id, ref_db_code, ref_db_id) 
+              VALUES (%s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/goa.py:80]:
```suggestion
        sql = """
            INSERT INTO protein2go 
                (protein_acc, term_id, ref_db_code, ref_db_id) 
            VALUES (%s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/goa.py [hunk 2]
```diff
@@ -74,8 +73,22 @@ def import_annotations(ora_url: str, pg_url: str):
...
+
+        for rec in ora_cur:
+            records.append(rec)
+            if len(records) == 1000:
+                pg_cur.executemany(sql, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/goa.py:88]:
```suggestion

        if records:
```

File: pyinterprod/pronto/goa.py [hunk 3]
```diff
@@ -95,8 +108,23 @@ def import_annotations(ora_url: str, pg_url: str):
...
             """
         )
-        sql = "INSERT INTO publication VALUES %s"
-        execute_values(pg_cur, sql, ora_cur, page_size=1000)
+
+        records = []
+        sql = """
+              INSERT INTO publication (id, title, published) 
+              VALUES (%s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/goa.py:116]:
```suggestion
        sql = """
            INSERT INTO publication (id, title, published) 
            VALUES (%s, %s, %s)
        """
```

File: pyinterprod/pronto/goa.py [hunk 4]
```diff
@@ -95,8 +108,23 @@ def import_annotations(ora_url: str, pg_url: str):
...
+
+        for rec in ora_cur:
+            records.append(rec)
+            if len(records) == 1000:
+                pg_cur.executemany(sql, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/goa.py:124]:
```suggestion

        if records:
```

File: pyinterprod/pronto/goa.py [hunk 5]
```diff
@@ -143,16 +171,32 @@ def import_annotations(ora_url: str, pg_url: str):
...
-            len(_get_constraints(row[0], ancestors, constraints)),
-            row[3] == 'Y',
-            row[4],
-            row[5]
-        ) for row in ora_cur), page_size=1000)
+        records = []
+        sql = """
+              INSERT INTO term (id, name, category, num_constraints, is_obsolete, definition, replaced_id) 
+              VALUES (%s, %s, %s, %s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/goa.py:178]:
```suggestion
        sql = """
            INSERT INTO term 
                (id, name, category, num_constraints, is_obsolete, definition, 
                replaced_id) 
            VALUES (%s, %s, %s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/goa.py [hunk 6]
```diff
@@ -143,16 +171,32 @@ def import_annotations(ora_url: str, pg_url: str):
...
+                    row[0],
+                    row[1],
+                    row[2],
+                    len(_get_constraints(row[0], ancestors, constraints)),
+                    row[3] == "Y",
+                    row[4],
+                    row[5],
+                )
+            )
+            if len(records) == 1000:
```
Comment (matthiasblum) [inline pyinterprod/pronto/goa.py:192]:
```suggestion

            if len(records) == 1000:
```

File: pyinterprod/pronto/match.py [hunk 7]
```diff
@@ -273,13 +272,26 @@ def insert_signature2protein(url: str, names_db: str, matches_file: str,
...
         proteins = _iter_proteins(names_db, matches_file, src, dst)
-        cur.copy_from(file=pg.CsvIO(proteins, sep='|'),
-                      table="signature2protein",
-                      sep='|')
-        con.commit()
+
+        sql = """
+              INSERT INTO signature2protein (signature_acc, model_acc, protein_acc, is_reviewed, taxon_left_num, name_id, md5) 
+              VALUES (%s, %s, %s, %s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/match.py:282]:
```suggestion
        sql = """
            INSERT INTO signature2protein 
                (signature_acc, model_acc, protein_acc, is_reviewed, 
                taxon_left_num, name_id, md5) 
            VALUES (%s, %s, %s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/match.py [hunk 8]
```diff
@@ -273,13 +272,26 @@ def insert_signature2protein(url: str, names_db: str, matches_file: str,
...
+
+        records = []
+        for row in proteins:
+            records.append(row)
+            if len(records) == 1000:
+                cur.executemany(sql, records)
+                con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/match.py:291]:
```suggestion

        if records:
```

File: pyinterprod/pronto/match.py [hunk 9]
```diff
@@ -467,22 +479,34 @@ def create_match_table(uri: str):
...
         name2id = dict(cur.fetchall())
 
-        cur.copy_from(file=pg.CsvIO(_get_fmatches(ora_uri, name2id), sep="|"),
-                      table="match",
-                      sep="|")
-        con.commit()
+        sql = """
+              INSERT INTO match (protein_acc, signature_acc, database_id, fragments) 
+              VALUES (%s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/match.py:490]:
```suggestion
        sql = """
            INSERT INTO match 
                (protein_acc, signature_acc, database_id, fragments) 
            VALUES (%s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/match.py [hunk 10]
```diff
@@ -467,22 +479,34 @@ def create_match_table(uri: str):
...
+
+        records = []
+        for row in _get_fmatches(ora_uri, name2id):
+            records.append(row)
+            if len(records) == 1000:
+                cur.executemany(sql, records)
+                con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/match.py:499]:
```suggestion

        if records:
```

File: pyinterprod/pronto/match.py [hunk 11]
```diff
@@ -545,16 +569,28 @@ def insert_matches(uri: str, matches_file: str, processes: int = 1):
...
         matches = _iter_matches(matches_file, name2id, src, dst)
-        cur.copy_from(file=pg.CsvIO(matches, sep="|"),
-                      table="match",
-                      sep="|")
-        con.commit()
+
+        sql = """
+              INSERT INTO match (protein_acc, signature_acc, database_id, fragments) 
+              VALUES (%s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/match.py:581]:
```suggestion
        sql = """
            INSERT INTO match 
                (protein_acc, signature_acc, database_id, fragments) 
            VALUES (%s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/match.py [hunk 12]
```diff
@@ -545,16 +569,28 @@ def insert_matches(uri: str, matches_file: str, processes: int = 1):
...
+
+        records = []
+        for row in matches:
+            records.append(row)
+            if len(records) == 1000:
+                cur.executemany(sql, records)
+                con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/match.py:590]:
```suggestion

        if records:
```

File: pyinterprod/pronto/protein.py [hunk 13]
```diff
@@ -54,8 +52,22 @@ def import_similarity_comments(swp_url: str, ipr_url: str):
             """
         )
 
-        sql = "INSERT INTO protein_similarity VALUES %s"
-        execute_values(pg_cur, sql, ora_cur, page_size=1000)
+        records = []
+        sql = """
+              INSERT INTO protein_similarity (comment_id, comment_text, protein_acc) 
+              VALUES (%s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:59]:
```suggestion
        sql = """
            INSERT INTO protein_similarity 
                (comment_id, comment_text, protein_acc) 
            VALUES (%s, %s, %s)
        """
```

File: pyinterprod/pronto/protein.py [hunk 14]
```diff
@@ -54,8 +52,22 @@ def import_similarity_comments(swp_url: str, ipr_url: str):
...
+
+        for rec in ora_cur:
+            records.append(rec)
+            if len(records) == 1000:
+                pg_cur.executemany(sql, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:67]:
```suggestion

        if records:
```

File: pyinterprod/pronto/protein.py [hunk 15]
```diff
@@ -142,7 +154,19 @@ def import_protein_names(swp_url: str, ipr_url: str, database: str,
...
+
+        sql_protein2name = """
+                          INSERT INTO protein2name (protein_acc, name_id) 
+                          VALUES (%s, %s)
+                          """
+
+        sql_protein_name = """
+                          INSERT INTO protein_name (name_id, text) 
+                          VALUES (%s, %s)
+                          """
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:168]:
```suggestion
        sql_protein2name = """
            INSERT INTO protein2name (protein_acc, name_id) 
            VALUES (%s, %s)
        """

        sql_protein_name = """
            INSERT INTO protein_name (name_id, text) 
            VALUES (%s, %s)
        """
```

File: pyinterprod/pronto/protein.py [hunk 16]
```diff
@@ -156,12 +180,17 @@ def import_protein_names(swp_url: str, ipr_url: str, database: str,
...
-                        argslist=values,
-                        page_size=1000
-                    )
+                    for rec in values:
+                        records.append(rec)
+                        if len(records) == 1000:
+                            pg_cur.executemany(sql_protein2name, records)
+                            pg_con.commit()
+                            records.clear()
+                    if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:189]:
```suggestion

                    if records:
```

File: pyinterprod/pronto/protein.py [hunk 17]
```diff
@@ -172,20 +201,29 @@ def import_protein_names(swp_url: str, ipr_url: str, database: str,
...
-                argslist=values,
-                page_size=1000
-            )
+            for rec in values:
+                records.append(rec)
+                if len(records) == 1000:
+                    pg_cur.executemany(sql_protein2name, records)
+                    pg_con.commit()
+                    records.clear()
+            if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:210]:
```suggestion

            if records:
```

File: pyinterprod/pronto/protein.py [hunk 18]
```diff
@@ -172,20 +201,29 @@ def import_protein_names(swp_url: str, ipr_url: str, database: str,
...
-            page_size=1000
-        )
+
+        for text, name_id in names.items():
+            records.append((name_id, text))
+            if len(records) == 1000:
+                pg_cur.executemany(sql_protein_name, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:223]:
```suggestion

        if records:
```

File: pyinterprod/pronto/protein.py [hunk 19]
```diff
@@ -196,14 +234,14 @@ def import_protein_names(swp_url: str, ipr_url: str, database: str,
 
     logger.info("copying database")
     shutil.copyfile(tmp_database, database)
-    logger.info(f"disk usage: {os.path.getsize(tmp_database)/1024**2:.0f} MB")
+    logger.info(f"disk usage: {os.path.getsize(tmp_database) / 1024 ** 2:.0f} MB")
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:237]:
```suggestion
    logger.info(f"disk usage: "
                f"{os.path.getsize(tmp_database) / 1024 ** 2:.0f} MB")
```

File: pyinterprod/pronto/protein.py [hunk 20]
```diff
@@ -220,25 +258,38 @@ def import_proteins(ora_url: str, pg_url: str):
...
-                ) for row in ora_cur)
-
-        pg_cur.copy_from(file=CsvIO(gen, sep='|'),
-                         table="protein",
-                         sep='|')
+
+        sql = """
+              INSERT INTO protein (accession, identifier, length, taxon_id, is_fragment, is_reviewed) 
+              VALUES (%s, %s, %s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:273]:
```suggestion
        sql = """
            INSERT INTO protein 
                (accession, identifier, length, taxon_id, is_fragment, 
                is_reviewed) 
            VALUES (%s, %s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/protein.py [hunk 21]
```diff
@@ -220,25 +258,38 @@ def import_proteins(ora_url: str, pg_url: str):
...
+                            row[2],
+                            row[3],
+                            row[4] == 'Y',
+                            row[5] == 'S'
+                            ))
+            if len(records) == 1000:
+                pg_cur.executemany(sql, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:288]:
```suggestion

        if records:
```

File: pyinterprod/pronto/protein.py [hunk 22]
```diff
@@ -220,25 +258,38 @@ def import_proteins(ora_url: str, pg_url: str):
...
+        records = []
+        for row in ora_cur:
+            records.append((row[0],
+                            row[1],
+                            row[2],
+                            row[3],
+                            row[4] == 'Y',
+                            row[5] == 'S'
+                            ))
+            if len(records) == 1000:
```
Comment (matthiasblum) [inline pyinterprod/pronto/protein.py:284]:
```suggestion

            if len(records) == 1000:
```

File: pyinterprod/pronto/proteome.py [hunk 23]
```diff
@@ -66,13 +65,40 @@ def import_proteomes(ora_url: str, pg_url: str):
...
         iterator = ProteomeIterator(ora_url)
-        execute_values(pg_cur, sql, iterator, page_size=1000)
 
-        sql = "INSERT INTO proteome VALUES %s"
-        execute_values(pg_cur, sql, list(iterator.proteomes.values()),
-                       page_size=1000)
+        sql = """
+              INSERT INTO proteome2protein (id, protein_acc) 
+              VALUES (%s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/proteome.py:74]:
```suggestion
        sql = """
            INSERT INTO proteome2protein (id, protein_acc) 
            VALUES (%s, %s)
        """
```

File: pyinterprod/pronto/proteome.py [hunk 24]
```diff
@@ -66,13 +65,40 @@ def import_proteomes(ora_url: str, pg_url: str):
...
+
+        for rec in iterator:
+            records.append(rec)
+            if len(records) == 1000:
+                pg_cur.executemany(sql, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/proteome.py:82]:
```suggestion

        if records:
```

File: pyinterprod/pronto/proteome.py [hunk 25]
```diff
@@ -66,13 +65,40 @@ def import_proteomes(ora_url: str, pg_url: str):
...
+              VALUES (%s, %s, %s, %s)
+              """
+
+        for rec in list(iterator.proteomes.values()):
+            records.append(rec)
+            if len(records) == 1000:
+                pg_cur.executemany(sql, records)
+                pg_con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/proteome.py:98]:
```suggestion

        if records:
```

File: pyinterprod/pronto/proteome.py [hunk 26]
```diff
@@ -66,13 +65,40 @@ def import_proteomes(ora_url: str, pg_url: str):
...
+            pg_cur.executemany(sql, records)
+            pg_con.commit()
+            records.clear()
+
+        sql = """
+              INSERT INTO proteome (id, name, taxon_id, num_proteins) 
+              VALUES (%s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/proteome.py:90]:
```suggestion
        sql = """
            INSERT INTO proteome (id, name, taxon_id, num_proteins) 
            VALUES (%s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/signature.py [hunk 27]
```diff
@@ -259,9 +258,25 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
         )
-        execute_values(cur, "INSERT INTO signature VALUES %s", values,
-                       page_size=1000)
-        con.commit()
+
+        records = []
+        sql = """
+              INSERT INTO signature (accession, database_id, name, description, type, abstract, num_sequences, num_reviewed_sequences, num_complete_sequences, num_complete_reviewed_sequences, num_residues) 
+              VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:266]:
```suggestion
        sql = """
            INSERT INTO signature 
                (accession, database_id, name, description, type, abstract, 
                num_sequences, num_reviewed_sequences, num_complete_sequences, 
                num_complete_reviewed_sequences, num_residues) 
            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/signature.py [hunk 28]
```diff
@@ -284,10 +299,22 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
         )
 
-        cur.copy_from(file=CsvIO(_iter_comparisons(comparisons), sep='|'),
-                      table="comparison",
-                      sep='|')
-        con.commit()
+        sql = """
+              INSERT INTO comparison (signature_acc_1, signature_acc_2, num_collocations, num_overlaps) 
+              VALUES (%s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:305]:
```suggestion
        sql = """
            INSERT INTO comparison 
                (signature_acc_1, signature_acc_2, num_collocations, 
                num_overlaps) 
            VALUES (%s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/signature.py [hunk 29]
```diff
@@ -284,10 +299,22 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
+
+        records = []
+        for row in _iter_comparisons(comparisons):
+            records.append(row)
+            if len(records) == 1000:
+                cur.executemany(sql, records)
+                con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:314]:
```suggestion

        if records:
```

File: pyinterprod/pronto/signature.py [hunk 30]
```diff
@@ -314,11 +341,22 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
 
-        records = _iter_predictions(comparisons, signatures)
-        cur.copy_from(file=CsvIO(records, sep='|'),
-                      table="prediction",
-                      sep='|')
-        con.commit()
+        sql = """
+              INSERT INTO prediction (signature_acc_1, signature_acc_2, num_collocations, num_protein_overlaps, num_residue_overlaps) 
+              VALUES (%s, %s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:347]:
```suggestion
        sql = """
            INSERT INTO prediction 
                (signature_acc_1, signature_acc_2, num_collocations, 
                num_protein_overlaps, num_residue_overlaps) 
            VALUES (%s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/signature.py [hunk 31]
```diff
@@ -314,11 +341,22 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
+
+        records = []
+        for row in _iter_predictions(comparisons, signatures):
+            records.append(row)
+            if len(records) == 1000:
+                cur.executemany(sql, records)
+                con.commit()
+                records.clear()
+        if records:
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:356]:
```suggestion

        if records:
```

File: pyinterprod/pronto/taxon.py [hunk 32]
```diff
@@ -103,12 +102,25 @@ def import_taxonomy(ora_url: str, pg_url: str):
...
-            (tax_id, name, rank, left_num, right_num, parent_id,
-             json.dumps(get_lineage(taxa, tax_id)))
-            for tax_id, (name, rank, left_num, right_num, parent_id)
-            in taxa.items()
-        ), page_size=1000)
+        records = []
+        sql = """
+              INSERT INTO taxon (id, name, rank, left_number, right_number, parent_id, lineage)
+              VALUES (%s, %s, %s, %s, %s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/taxon.py:109]:
```suggestion
        sql = """
            INSERT INTO taxon 
                (id, name, rank, left_number, right_number, parent_id, lineage)
            VALUES (%s, %s, %s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/taxon.py [hunk 33]
```diff
@@ -103,12 +102,25 @@ def import_taxonomy(ora_url: str, pg_url: str):
...
+
+        for tax_id, (name, rank, left_num, right_num, parent_id) in taxa.items():
+            json_lineage = json.dumps(get_lineage(taxa, tax_id))
+            records.append((tax_id, name, rank, left_num, right_num, parent_id, json_lineage))
```
Comment (matthiasblum) [inline pyinterprod/pronto/taxon.py:113]:
```suggestion
        for tax_id, value in taxa.items():
            (name, rank, left_num, right_num, parent_id) = value
            json_lineage = json.dumps(get_lineage(taxa, tax_id))
            records.append((tax_id, name, rank, left_num,
                            right_num, parent_id, json_lineage))
```

File: pyinterprod/pronto/taxon.py [hunk 34]
```diff
@@ -118,9 +130,23 @@ def import_taxonomy(ora_url: str, pg_url: str):
         )
 
         logger.info("populating: lineage")
-        execute_values(pg_cur, "INSERT INTO lineage VALUES %s",
-                       iter_lineage(taxa),
-                       page_size=1000)
+        sql = """
+              INSERT INTO lineage (child_id, parent_id, parent_rank) 
+              VALUES (%s, %s, %s)
+              """
```
Comment (matthiasblum) [inline pyinterprod/pronto/taxon.py:136]:
```suggestion
        sql = """
            INSERT INTO lineage (child_id, parent_id, parent_rank) 
            VALUES (%s, %s, %s)
        """
```

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Thank you. I've added a few suggestions, mostly to comply with PEP-8 (max 80 characters/line).

It seems that the [CsvIO class](https://github.com/ProteinsWebTeam/pyinterprod/blob/95225efc686b38539c8bb31d7daa14ed5d667a34/pyinterprod/utils/pg.py#L8) is not required anymore. Please delete the implementation and imports.

Finally, can you post the run time of Pronto tasks so we can compare performances?

**Comment by laisecf** [comment]
> Thank you. I've added a few suggestions, mostly to comply with PEP-8 (max 80 characters/line).
> 
> It seems that the [CsvIO class](https://github.com/ProteinsWebTeam/pyinterprod/blob/95225efc686b38539c8bb31d7daa14ed5d667a34/pyinterprod/utils/pg.py#L8) is not required anymore. Please delete the implementation and imports.
> 
> Finally, can you post the run time of Pronto tasks so we can compare performances?


**The run with the new dependences takes more ~10h30**

**The last production run:**
Screenshot 2023-07-17 at 16 59 32
                                                                                                                                         
pronto-annotations                   07 17:55:09    07 18:27:35    ~30  
pronto-taxonomy                      07 17:55:09    07 17:59:51    4min 
pronto-databases                     07 17:55:09    07 17:55:20     0   
pronto-export-matches                07 17:55:10    08 03:54:16    ~10h 
pronto-proteins-similarities         07 17:55:10    07 18:03:54    8min 
pronto-proteomes                     07 17:55:10    07 18:26:10    ~30mi
pronto-proteins                      07 17:55:10    07 18:57:35    ~1h  
pronto-init-matches                  07 17:55:10    07 17:55:17     0   
pronto-proteins-names                07 17:55:10    07 21:57:07    ~4h  
pronto-insert-fmatches               07 17:55:41    07 18:11:41    16min
pronto-insert-matches                08 03:54:32    08 04:16:14    22min
pronto-signatures                    08 03:54:32    08 05:17:50    ~1h30
pronto-insert-signature2proteins     08 03:54:32    08 04:40:17    ~1h  
pronto-index-matches                 08 04:16:20    08 08:42:47    ~4h30
pronto-index-signature2proteins      09 12:54:10    09 19:49:59    ~7h  

**The test with dependencies updated:**
Screenshot 2023-07-17 at 16 57 43
                                                            
pronto-annotations                   10 16:44:43    10 17:05:13   20min 
pronto-taxonomy                      10 16:44:43    10 16:52:55    8min 
pronto-databases                     10 16:44:44    10 16:44:49     0   
pronto-export-matches                10 16:44:44    11 02:44:01    ~10h 
pronto-proteins-similarities         10 16:44:44    10 16:49:24    5min 
pronto-proteomes                     10 16:44:44    10 17:49:49    ~1h  
pronto-proteins                      10 16:44:44    10 19:20:29    ~3h  
pronto-init-matches                  10 16:44:44    10 16:45:08     0   
pronto-proteins-names                10 16:44:44    10 20:48:29    ~4h  
pronto-insert-fmatches               10 16:45:14    10 18:02:29    ~1h  
pronto-insert-matches                11 02:44:05    11 05:34:52    ~3h  
pronto-signatures                    11 02:44:06    11 04:21:19    ~2h  
pronto-insert-signature2proteins     11 02:44:06    11 05:56:30    ~3h  
pronto-index-matches                 11 05:34:59    11 12:02:39    ~7h  
pronto-index-signature2proteins      11 05:56:34    11 13:15:23    ~7h

**Comment by laisecf** [comment]
> Thank you. I've added a few suggestions, mostly to comply with PEP-8 (max 80 characters/line).
> 
> It seems that the [CsvIO class](https://github.com/ProteinsWebTeam/pyinterprod/blob/95225efc686b38539c8bb31d7daa14ed5d667a34/pyinterprod/utils/pg.py#L8) is not required anymore. Please delete the implementation and imports.
> 
> Finally, can you post the run time of Pronto tasks so we can compare performances?

The run with the executemany takes plus ~10h30

| Task Name                        | Runtime before           | Runtime after            | Difference |
|----------------------------------|--------------------------|--------------------------|------------|
| pronto-annotations               | 07/17:55:09->07/18:27:35 | 10/16:44:43->10/17:05:13 | -          |
| pronto-taxonomy                  | 07/17:55:09->07/17:59:51 | 10/16:44:43->10/16:52:55 | -          |
| pronto-databases                 | 07/17:55:09->07/17:55:20 | 10/16:44:44->10/16:44:49 | -          |
| pronto-export-matches            | 07/17:55:10->08/03:54:16 | 10/16:44:44->11/02:44:01 | -          |
| pronto-proteins-similarities     | 07/17:55:10->07/18:03:54 | 10/16:44:44->10/16:49:24 | -          |
| pronto-proteomes                 | 07/17:55:10->07/18:26:10 | 10/16:44:44->10/17:49:49 | ~30min     |
| pronto-proteins                  | 07/17:55:10->07/18:57:35 | 10/16:44:44->10/19:20:29 | ~2h        |
| pronto-init-matches              | 07/17:55:10->07/17:55:17 | 10/16:44:44->10/16:45:08 | -          |
| pronto-proteins-names            | 07/17:55:10->07/21:57:07 | 10/16:44:44->10/20:48:29 | -          |
| pronto-insert-fmatches           | 07/17:55:41->07/18:11:41 | 10/16:45:14->10/18:02:29 | ~45min     |
| pronto-insert-matches            | 08/03:54:32->08/04:16:14 | 11/02:44:05->11/05:34:52 | ~2h30      |
| pronto-signatures                | 08/03:54:32->08/05:17:50 | 11/02:44:06->11/04:21:19 | ~30min     |
| pronto-insert-signature2proteins | 08/03:54:32->08/04:40:17 | 11/02:44:06->11/05:56:30 | ~2h        |
| pronto-index-matches             | 08/04:16:20->08/08:42:47 | 11/05:34:59->11/12:02:39 | ~2h30      |
| pronto-index-signature2proteins  | 09/12:54:10->09/19:49:59 | 11/05:56:34->11/13:15:23 | -          |

**Comment by laisecf** [comment]
> > Thank you. I've added a few suggestions, mostly to comply with PEP-8 (max 80 characters/line).
> > It seems that the [CsvIO class](https://github.com/ProteinsWebTeam/pyinterprod/blob/95225efc686b38539c8bb31d7daa14ed5d667a34/pyinterprod/utils/pg.py#L8) is not required anymore. Please delete the implementation and imports.
> > Finally, can you post the run time of Pronto tasks so we can compare performances?
> 
> The run with the executemany takes plus ~10h30
> 
> Task Name	Runtime before	Runtime after	Difference
> pronto-annotations	07/17:55:09->07/18:27:35	10/16:44:43->10/17:05:13	-
> pronto-taxonomy	07/17:55:09->07/17:59:51	10/16:44:43->10/16:52:55	-
> pronto-databases	07/17:55:09->07/17:55:20	10/16:44:44->10/16:44:49	-
> pronto-export-matches	07/17:55:10->08/03:54:16	10/16:44:44->11/02:44:01	-
> pronto-proteins-similarities	07/17:55:10->07/18:03:54	10/16:44:44->10/16:49:24	-
> pronto-proteomes	07/17:55:10->07/18:26:10	10/16:44:44->10/17:49:49	~30min
> pronto-proteins	07/17:55:10->07/18:57:35	10/16:44:44->10/19:20:29	~2h
> pronto-init-matches	07/17:55:10->07/17:55:17	10/16:44:44->10/16:45:08	-
> pronto-proteins-names	07/17:55:10->07/21:57:07	10/16:44:44->10/20:48:29	-
> pronto-insert-fmatches	07/17:55:41->07/18:11:41	10/16:45:14->10/18:02:29	~45min
> pronto-insert-matches	08/03:54:32->08/04:16:14	11/02:44:05->11/05:34:52	~2h30
> pronto-signatures	08/03:54:32->08/05:17:50	11/02:44:06->11/04:21:19	~30min
> pronto-insert-signature2proteins	08/03:54:32->08/04:40:17	11/02:44:06->11/05:56:30	~2h
> pronto-index-matches	08/04:16:20->08/08:42:47	11/05:34:59->11/12:02:39	~2h30
> pronto-index-signature2proteins	09/12:54:10->09/19:49:59	11/05:56:34->11/13:15:23	-

### Considering just pronto tasks that had time difference:
 
**Tasks I didn't change to copy (still executemany because is a little time diff)**
- pronto-proteomes                                 23 22:38:32      23 23:27:55    ~50min
- pronto-signatures                                  24 06:28:28     24 07:58:49    ~1h30

**Changing to copy (improved a lot the time, practically the same as before the dependences update)**
- pronto-proteins                                      24 10:17:58      24 11:37:39     ~1h20   
- pronto-insert-fmatches                         23 22:39:02     23 23:21:45    ~45min
- pronto-insert-matches                          24 06:28:28     24 06:57:48    ~30min
- pronto-insert-signature2proteins         24 06:28:29     24 07:00:33    ~30min
- pronto-index-matches                           24 06:57:52     24 07:26:05    ~30min

File: pyinterprod/pronto/proteome.py [hunk 35]
```diff
@@ -66,13 +65,42 @@ def import_proteomes(ora_url: str, pg_url: str):
...
+            pg_cur.executemany(sql, records)
+            pg_con.commit()
+            records.clear()
+
+        sql = """
+            INSERT INTO proteome (id, name, taxon_id, num_proteins) 
+            VALUES (%s, %s, %s, %s)
+        """
+
+        for rec in list(iterator.proteomes.values()):
```
Comment (matthiasblum) [inline pyinterprod/pronto/proteome.py:93]:
```suggestion
        for rec in iterator.proteomes.values():
```

File: pyinterprod/pronto/signature.py [hunk 36]
```diff
@@ -259,9 +258,28 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
-                       page_size=1000)
-        con.commit()
+
+        sql = """
+                    INSERT INTO signature 
+                        (accession, database_id, name, description, type, abstract, 
+                        num_sequences, num_reviewed_sequences, num_complete_sequences, 
+                        num_complete_reviewed_sequences, num_residues) 
+                    VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+                """
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:268]:
```suggestion
        sql = """
            INSERT INTO signature (
                accession, database_id, name, description, type, abstract, 
                num_sequences, num_reviewed_sequences, num_complete_sequences, 
                num_complete_reviewed_sequences, num_residues
            ) 
            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/signature.py [hunk 37]
```diff
@@ -284,10 +302,25 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
-        cur.copy_from(file=CsvIO(_iter_comparisons(comparisons), sep='|'),
-                      table="comparison",
-                      sep='|')
-        con.commit()
+        sql = """
+                    INSERT INTO comparison 
+                        (signature_acc_1, signature_acc_2, num_collocations, 
+                        num_overlaps) 
+                    VALUES (%s, %s, %s, %s)
+                """
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:310]:
```suggestion
        sql = """
            INSERT INTO comparison (signature_acc_1, signature_acc_2, 
                                    num_collocations, num_overlaps) 
            VALUES (%s, %s, %s, %s)
        """
```

File: pyinterprod/pronto/signature.py [hunk 38]
```diff
@@ -314,11 +347,25 @@ def insert_signatures(ora_uri: str, pg_uri: str, matches_file: str,
...
-        cur.copy_from(file=CsvIO(records, sep='|'),
-                      table="prediction",
-                      sep='|')
-        con.commit()
+        sql = """
+                    INSERT INTO prediction 
+                        (signature_acc_1, signature_acc_2, num_collocations, 
+                        num_protein_overlaps, num_residue_overlaps) 
+                    VALUES (%s, %s, %s, %s, %s)
+                """
```
Comment (matthiasblum) [inline pyinterprod/pronto/signature.py:355]:
```suggestion
        sql = """
            INSERT INTO prediction (
                signature_acc_1, signature_acc_2, num_collocations, 
                num_protein_overlaps, num_residue_overlaps
            ) 
            VALUES (%s, %s, %s, %s, %s)
        """
```

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Looks good. I made a few suggestions, but they are mostly indentation-related. Thanks!

**Comment by laisecf** [comment]
> Looks good. I made a few suggestions, but they are mostly indentation-related. Thanks!

Done! And sorry, I will remember to check indentation before commit

**Review by matthiasblum — APPROVED** [review]

**Pull request merged**